kratos/tests/test_processes.py

```diff
@@ -2521,6 +2521,84 @@ def test_split_internal_interfaces_process_3D(self):
+                        "python_module": "assign_mpcs_to_neighbours_process",
+                        "kratos_module": "KratosMultiphysics",
+                        "process_name": "AssignMPCsToNeighboursProcess",
+                        "Parameters": {
+                            "computing_model_part_name": "Main",
+                            "master_model_part_name": "Main.GENERIC_Master",
+                            "slave_model_part_name": "Main.GENERIC_Slave",
+                            "search_radius": 0.01,
+                            "minimum_number_of_neighbouring_nodes": 3,
+                            "assign_mpcs_every_time_step":  true,
+                            "variable_names": ["DISPLACEMENT_X"]
+                        }
+                    }
+                    ]
+                }
+            """
+        )
+
+        # Create and execute processes
+        list_of_processes = process_factory.KratosProcessFactory(current_model).ConstructListOfProcesses(settings["process_list"])
+        for process in list_of_processes:
+            process.ExecuteInitialize()
+            process.ExecuteInitializeSolutionStep()
+
+        # Define expected constraint weights
+        expected_constraints_weights = [
+            [7.894583578700136e-06, 0.9999842108328428, 7.89458357849079e-06],
+            [1.5078135870940287e-06, 6.621797865852621e-06, 0.9999918703885471],
+            [0.999991870388547, 6.6217978659788195e-06, 1.5078135870889554e-06],
+        ]
+
+        # Obtain constraint weights from the model
+        obtained_constraints_weights = []
+        contraints_to_validate = [1, 2, 3]
+        for i, id in enumerate(contraints_to_validate):
+            const = model_part.GetMasterSlaveConstraint(id)
+            output_contraints_weights = KratosMultiphysics.Matrix()
+            constants = KratosMultiphysics.Vector()
+            const.CalculateLocalSystem(output_contraints_weights, constants, KratosMultiphysics.ProcessInfo())
+
+            obtained_weights = [output_contraints_weights[0, j] for j in range(3)]
+            obtained_constraints_weights.append(obtained_weights)
+
+        # Define a function to check if two weight lists match within a specified tolerance
+        def check_matching_weights(expected_weights, obtained_weights, delta=1e-6):
+            return all(abs(ew - ow) <= delta for ew, ow in zip(expected_weights, obtained_weights))
+
+        # Compare each expected_weights to obtained_weights, ensuring that there is a match
+        for expected_weights in expected_constraints_weights:
+            matching_found = False
+
+            for obtained_weights in obtained_constraints_weights:
+                if check_matching_weights(expected_weights, obtained_weights):
+                    for ew, ow in zip(expected_weights, obtained_weights):
+                        self.assertAlmostEqual(ew, ow, delta=1e-6)  # Assert the weights are equal within the specified tolerance

```

Comment:
If I'm not wrong, there is an `assertAlmostEqualVector` or something of this sort.

Reply:
assertVectorAlmostEqual